### ISSUE TITLE
Mention version support for all building blocks

### ISSUE THREAD

**Issue by pablopaul** [zurb/building-blocks#211]
Each building block should mention when it needs flexbox mode to work correctly (since flexbox mode is not the default).

Successor of #208

**Comment by IamManchanda** [comment]
cc / @rafibomb !

**Comment by kball** [comment]
Hey guys - we've just pushed up some work in progress to make this happen, but could use some help in adding requirements to all of the existing building blocks.

I just pushed this up in this branch: https://github.com/zurb/building-blocks/tree/add-dependency-info

This updates the build to look for a `requirements` array in the building block yml file (see https://github.com/zurb/building-blocks/commit/b1176431ae59c981f5f975f883196871ca00b040#diff-36cd369ab7ae017d815e6da691e6bc97R21)

To get this to become a reality, we need to add these requirements to the building blocks that need them. Can I recruit help to get those into the yml files?  To prevent duplicating work, maybe folks can claim categories and add requirements as needed within a category, as PRs aimed against the `add-dependency-info` branch?

CC @pablopaul @IamManchanda @rafibomb

**Comment by IamManchanda** [comment]
@kball before starting ... I think we should get codepen example also in there same like docs?? 
What you guys say??

**Comment by pablopaul** [comment]
@IamManchanda I think a codepen example is another Issue and no requirement for the new requirements block

@kball Thanks for the first step to make `requirements` for BB reality. Can you give potential helping folks a short summary what requirements could be possible besides flexbox?

**Comment by IamManchanda** [comment]
@pablopaul .... i already have so many created threads (https://git.io/vHfsM) .... i dont think i need to create another .... :smile: 
I will get my answer here only i guess .... what you think @kball ..... i think we should have codepen examples too ..... 
Secondly, i think it can be helpful to find out on the building block healthchecks with both grids (flex, float) by codepen example .... so there also its a win win situation!

**Comment by kball** [comment]
@pablopaul flexbox and css-grid are the 2 main ones that come to mind... maybe also fontawesome (since we use that in many places and it hasn't always been clear to folks).

Longer term, with some additional work, this may open the door to adding dependencies on other 3rd party packages, but for first version probably not.

**Comment by IamManchanda** [comment]
@kball if the block works with both the grids (flex, float) then what to write ?? 
For eg => [List Group component](http://foundation.zurb.com/building-blocks/blocks/list-group.html)

**Comment by kball** [comment]
@IamManchanda if it works with both, and does not itself use flex attributes, no need to have a requirement. If it *does* have flex attributes, please add flexbox requirement

**Comment by IamManchanda** [comment]
Ohk but what about codepens ?? :smile:

**Comment by kball** [comment]
@IamManchanda codepens are a good idea, especially now that we figured out how to do the scss dependency stuff in codepen for the docs.  That said, I think that is it's own (moderately large) push.

Our next iteration on the docs will be to do an integration of related building blocks, so perhaps codepens in BBs can be a part of that push, as that will lead to a more unified feeling experience. I'll talk about this with @jkfcheng who has been the main designer on the building blocks site... for now, lets focus on knocking out the tech requirements. :)

**Comment by IamManchanda** [comment]
Ohk ohk... Assign me categories (any u would like)  and I will do! 

Secondly, I think we should add font awesome as requirement!

**Comment by pablopaul** [comment]
@kball @IamManchanda the BB categories are:

* Kits
* Navigation -> @pablopaul
* Status
* Media
* Containers
* Forms
* Control

I will start with Navigation BB´s, @IamManchanda how about to start with Kits?

**Comment by pablopaul** [comment]
@kball `font-awesome` should be definitely mentioned as requirement when used in BB´s.

**Comment by IamManchanda** [comment]
@pablopaul kits are combo of the other categories only .... so that's not basically a category!
Kits are a same type of building block (eg.. Blogs Kit)

Besides that,
I have added my assigned category with a checklist in your above comment 
But will wait for @kball for revert on font awesome 
As i dont want to do reworks :smile: .... even if its just find all replace all :wink:

**Comment by kball** [comment]
commented on font awesome on the other issue, but generally yes mention it as a dependency. 

So that it doesn't feel completely out of the blue/leaving folks stranded, before this goes live I want to add "about" or "how to" pages as urls that these link to (you can see the beginnings of this at https://github.com/zurb/building-blocks/commit/b1176431ae59c981f5f975f883196871ca00b040#diff-1f393f5eed2ff868ac1704ac692bd5d0R36

This will probably be a global datafile mapping dependency names to descriptive urls.

**Comment by kball** [comment]
motion-ui is another dependency to watch for. Noticed some BBs assume it's there.

**Comment by pablopaul** [comment]
@kball I thought we just mention requirements which are not included in the starter projects, `motion-ui` is included by default. Under this assumption, the `flexbox` requirement will become obsolete when it becomes the default mode in grid. 

So we also include requirements which are activated in starter projects, correct?

**Comment by kball** [comment]
So the thing I was thinking is... `motion-ui` is technically a separate package, so if it's required for functioning it should be in there; not everyone uses the ZURB stack and has it added by default.  As we move forward with building blocks, I actually imagine this expanding to allow more extensive package dependencies (e.g. building on top of independent JS packages)

With regards to flexbox, since the 6 series gives you an option to run without flexbox to support IE9 and older Android, we should highlight it through the duration of Foundation 6... I think once Foundation 7 ships and we drop support for those older browsers, then we can seriously talk about removing...

Along with https://github.com/zurb/building-blocks/issues/225 this should make it clear both which components from Foundation you need and which external techs you need to be allowing.

**Comment by IamManchanda** [comment]
Hey guys sorry to be left out with the going process but I am working on menu right now 
So I won't be able to contribute until weekend. 

That being said I will surely contribute this weekend on this! 
I hope this is not urgent.

**Comment by rafibomb** [comment]
Ok - yes - MotionUI is a separate library but it is included by default in the CSS version through the ZURB stack, so might create confusion to see it required or maybe we add tooltips to mention they likely already have it. jQuery would then fall into the same category.

Agree with @kball on Flex mode - think flexbox needs to be called out (most of the BB's are flex). But would be nice to test some of these to see if they are similar in both and label them as such.

So the list I have in my head - 
- **fontawesome** (if used) and mention that any icon or image can be used in it's place
- **flex mode** 

Can't think of any others right now... any more you all see?

**Comment by kball** [comment]
I think there's a distinction between jQuery (*required* to run Foundation JS) and MotionUI (billed as an optional dependency)

**Comment by rafibomb** [comment]
ok to add - MotionUI with a note that it's included with Foundation
